src/openrct2/object/ObjectList.cpp

```diff
@@ -170,12 +170,14 @@ void ObjectList::SetObject(ObjectType type, ObjectEntryIndex index, std::string_
     SetObject(index, entry);
 }
 
-ObjectEntryIndex ObjectList::Find(ObjectType type, std::string_view identifier)
+ObjectEntryIndex ObjectList::Find(ObjectType type, std::string_view identifier) const
 {
     auto& subList = GetList(type);
     for (size_t i = 0; i < subList.size(); i++)
     {
-        if (subList[i].Identifier == identifier)
+        auto& entry = subList[i];
+        if ((entry.Generation == ObjectGeneration::JSON && entry.Identifier == identifier)
+            || (entry.Generation == ObjectGeneration::DAT && entry.Entry.GetName() == identifier))
```

Comment:
This is not safe to do. The name is not unique for a dat. You should use the full identifier when comparing dats.

Reply:
Hmm, you mean including the checksum? That's unfortunate. My intended use case is to quickly see if particular vanilla scenery group objects have been loaded.

Do you think it'd be acceptable to add a `FindLegacyByName` method that performs this particular DAT check instead?